doc: add some info about nix shells in hacking.rst (#7295)
ocaml/dune: doc/hacking.rst

This profile might need to be updated from time to time, since the bootstrapped
version of Dune may become stale. This can be done by running the first command.

We have the following shells for specific tasks:

> - `nix develop .#slim` for a dev environment with fewer dependencies 

Author: ```suggestion

- `nix develop .#slim` for a dev environment with fewer dependencies 
```

> - You may also use `nix develop .#slim` for a dev environment with less
>   dependencies that is faster to build.

Reviewer: ```suggestion
- `nix develop .#slim` for a dev environment with fewer dependencies 
   that is faster to build.
```
